File: syntaxes/pygments/syntax.py
from pygments.lexer import RegexLexer, bygroups
from pygments.token import *

import re

__all__=['KyomlLexer']

class KyomlLexer(RegexLexer):
    name = 'Kyoml'
    aliases = ['kyoml']
    filenames = ['*.kyo', '*.kyoml']
    flags = re.MULTILINE | re.UNICODE

    tokens = {
        'root' : [
            (u'([-+]?[\\d.]+)', bygroups(Number)),
            (u'(\\{)', bygroups(Punctuation), 'main__1'),
            (u'(\\b[a-zA-Z][a-zA-Z0-9_]*)', bygroups(Keyword)),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*)\\()', bygroups(Name.Function), 'main__2'),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*))', bygroups(Name.Function)),
            (u'([=:]|\\|>|<\\|)', bygroups(Operator)),
            (u'([\\[])', bygroups(Punctuation), 'main__3'),
            (u'(,)', bygroups(Punctuation)),
            (u'(\\\")', bygroups(String), 'main__4'),
            (u'(\\\')', bygroups(String), 'main__5'),
            (u'(#.*)', bygroups(Comment)),
            (u'([^\\s\\n\\r])', bygroups(Generic.Error)),
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ], 
        'interpolation__1' : [
            ('(\n|\r|\r\n)', String),
            ('.', String.Interpol),
        ], 
        'main__1' : [
            (u'([-+]?[\\d.]+)', bygroups(Number)),
            (u'(\\{)', bygroups(Punctuation), 'main__1'),
            (u'(\\b[a-zA-Z][a-zA-Z0-9_]*)', bygroups(Keyword)),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*)\\()', bygroups(Name.Function), 'main__2'),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*))', bygroups(Name.Function)),
            (u'([=:]|\\|>|<\\|)', bygroups(Operator)),
            (u'([\\[])', bygroups(Punctuation), 'main__3'),
            (u'(,)', bygroups(Punctuation)),
            (u'(\\\")', bygroups(String), 'main__4'),
            (u'(\\\')', bygroups(String), 'main__5'),
            (u'(#.*)', bygroups(Comment)),
            (u'([^\\s\\n\\r])', bygroups(Generic.Error)),
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ], 
        'main__2' : [
            (u'([-+]?[\\d.]+)', bygroups(Number)),
            (u'(\\{)', bygroups(Punctuation), 'main__1'),
            (u'(\\b[a-zA-Z][a-zA-Z0-9_]*)', bygroups(Keyword)),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*)\\()', bygroups(Name.Function), 'main__2'),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*))', bygroups(Name.Function)),
            (u'([=:]|\\|>|<\\|)', bygroups(Operator)),
            (u'([\\[])', bygroups(Punctuation), 'main__3'),
            (u'(,)', bygroups(Punctuation)),
            (u'(\\\")', bygroups(String), 'main__4'),
            (u'(\\\')', bygroups(String), 'main__5'),
            (u'(#.*)', bygroups(Comment)),
            (u'([^\\s\\n\\r])', bygroups(Generic.Error)),
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ], 
        'main__3' : [
            (u'([-+]?[\\d.]+)', bygroups(Number)),
            (u'(\\{)', bygroups(Punctuation), 'main__1'),
            (u'(\\b[a-zA-Z][a-zA-Z0-9_]*)', bygroups(Keyword)),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*)\\()', bygroups(Name.Function), 'main__2'),
            (u'(@(\\b[a-zA-Z][a-zA-Z0-9_]*))', bygroups(Name.Function)),
            (u'([=:]|\\|>|<\\|)', bygroups(Operator)),
            (u'([\\[])', bygroups(Punctuation), 'main__3'),
            (u'(,)', bygroups(Punctuation)),
            (u'(\\\")', bygroups(String), 'main__4'),
            (u'(\\\')', bygroups(String), 'main__5'),
            (u'(#.*)', bygroups(Comment)),
            (u'([^\\s\\n\\r])', bygroups(Generic.Error)),
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ], 
        'main__4' : [
            (u'(\\$\\{)', bygroups(String.Interpol), 'interpolation__1'),
            (u'(.)', bygroups(String)),
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ], 
        'main__5' : [
            ('(\n|\r|\r\n)', String),
            ('.', String),
        ]
    }

File: syntaxes/rouge/syntax.rb
# -*- coding: utf-8 -*- #

module Rouge
    module Lexers
      class Kyoml < RegexLexer
        title     "kyoml"
        tag       'Kyoml'
        mimetypes 'text/x-kyoml'
        filenames '*.kyo', '*.kyoml'
  
        state:root do
            rule /([-+]?[\d.]+)/, Number
            rule /(\{)/, Punctuation, :main__1
            rule /(\b[a-zA-Z][a-zA-Z0-9_]*)/, Keyword
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*)\()/, Name::Function, :main__2
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*))/, Name::Function
            rule /([=:]|\|>|<\|)/, Operator
            rule /([\[])/, Punctuation, :main__3
            rule /(,)/, Punctuation
            rule /(\")/, String, :main__4
            rule /(\')/, String, :main__5
            rule /(#.*)/, Comment
            rule /([^\s\n\r]/, Generic::Error
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
        state:interpolation__1 do
            rule /(\n|\r|\r\n)/, String
            rule /./, String.Interpol
        end
  
        state:main__1 do
            rule /([-+]?[\d.]+)/, Number
            rule /(\{)/, Punctuation, :main__1
            rule /(\b[a-zA-Z][a-zA-Z0-9_]*)/, Keyword
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*)\()/, Name::Function, :main__2
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*))/, Name::Function
            rule /([=:]|\|>|<\|)/, Operator
            rule /([\[])/, Punctuation, :main__3
            rule /(,)/, Punctuation
            rule /(\")/, String, :main__4
            rule /(\')/, String, :main__5
            rule /(#.*)/, Comment
            rule /([^\s\n\r]/, Generic::Error
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
        state:main__2 do
            rule /([-+]?[\d.]+)/, Number
            rule /(\{)/, Punctuation, :main__1
            rule /(\b[a-zA-Z][a-zA-Z0-9_]*)/, Keyword
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*)\()/, Name::Function, :main__2
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*))/, Name::Function
            rule /([=:]|\|>|<\|)/, Operator
            rule /([\[])/, Punctuation, :main__3
            rule /(,)/, Punctuation
            rule /(\")/, String, :main__4
            rule /(\')/, String, :main__5
            rule /(#.*)/, Comment
            rule /([^\s\n\r]/, Generic::Error
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
        state:main__3 do
            rule /([-+]?[\d.]+)/, Number
            rule /(\{)/, Punctuation, :main__1
            rule /(\b[a-zA-Z][a-zA-Z0-9_]*)/, Keyword
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*)\()/, Name::Function, :main__2
            rule /(@(\b[a-zA-Z][a-zA-Z0-9_]*))/, Name::Function
            rule /([=:]|\|>|<\|)/, Operator
            rule /([\[])/, Punctuation, :main__3
            rule /(,)/, Punctuation
            rule /(\")/, String, :main__4
            rule /(\')/, String, :main__5
            rule /(#.*)/, Comment
            rule /([^\s\n\r]/, Generic::Error
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
        state:main__4 do
            rule /(\$\{)/, String::Interpol, :interpolation__1
            rule /(.)/, String
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
        state:main__5 do
            rule /(\n|\r|\r\n)/, String
            rule /./, String
        end
  
      end
    end
  end
  
  